src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferService.java

```diff
@@ -1,51 +1,327 @@
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .error(errorResponse)
+              .resp(responseRequestId)
+              .build();
+    }
+    log.info(hipHealthInformationRequestAcknowledgement.toString());
+    try {
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(
+              dataOnRequestPath, hipHealthInformationRequestAcknowledgement);
+      log.debug(dataOnRequestPath + " : dataOnRequest: " + response.getStatusCode());
+      if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_SUCCESS,
+            hipHealthInformationRequestAcknowledgement);
+      } else if (Objects.nonNull(response.getBody())
+          && Objects.nonNull(response.getBody().getErrorResponse())) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_ERROR,
+            hipHealthInformationRequestAcknowledgement);
+      }
+    } catch (Exception ex) {
+      String error =
+          "An unknown error occurred while calling Gateway API: "
+              + ex.getMessage()
+              + " unwrapped exception: "
+              + Exceptions.unwrap(ex);
+      log.debug(error);
+    }
   }
 
   /**
    * Requesting HIP for FHIR bundle
-   * @param HIPHealthInformationRequest use the requestId to fetch the careContexts from dump to request HIP.
+   *
+   * @param hipHealthInformationRequest use the requestId to fetch the careContexts from dump to
+   *     request HIP.
    */
   @Override
-  public void initiateBundleRequest(HIPHealthInformationRequest HIPHealthInformationRequest) {
-    // TODO Implement bundle request to HIP workflow logic.
+  public void initiateBundleRequest(HIPHealthInformationRequest hipHealthInformationRequest)
+      throws IllegalDataStateException {
+    RequestLog existingLog =
+        logsRepo.findByConsentId(hipHealthInformationRequest.getHiRequest().getConsent().getId());
+    if (existingLog == null) {
+      throw new IllegalDataStateException("consent id not found in db");
+    }
+    HIPNotifyRequest hipNotifyRequest =
+        (HIPNotifyRequest)
+            existingLog.getRequestDetails().get(FieldIdentifiers.DATA_NOTIFY_REQUEST);
+    HIPRequestBundle hipRequestBundle =
+        HIPRequestBundle.builder()
+            .careContextsWithPatientReferences(
+                hipNotifyRequest.getNotification().getConsentDetail().getCareContexts())
+            .consentId(hipHealthInformationRequest.getHiRequest().getConsent().getId())
+            .build();
+    try {
+      ResponseEntity<GatewayGenericResponse> response =
```

Comment:
this request is not going to gateway so can we handle this separately like in HIPClient and with appropriate response as well like ResourceBundleResponse